File: frontend/relay-frontend/src/app/models/camera-models.ts
export interface CameraSummary {
  id: string;
  name: string;
  host: string;
  port: number;
}

export interface CameraStatus {
  cameraId: string;
  reachable: boolean;
  statusCode: number;
  body: string;
  observedAt: string;
}

export interface SessionState {
  cameraId: string;
  active: boolean;
  changedAt: string;
}

export interface FramePayload {
  cameraId: string;
  contentType: string;
  data: number[];
  observedAt: string;
}

File: relay-backend/src/main/java/com/runtracer/relaybackend/camera/CameraModels.java
package com.runtracer.relaybackend.camera;

import java.time.Instant;

public class CameraModels {
    public record CameraSummary(String id, String name, String host, int port, String uri) {}
    public record CameraStatus(String cameraId, boolean reachable, int statusCode, String body, Instant observedAt) {}
    public record SessionState(String cameraId, boolean active, Instant changedAt) {}
    public record FramePayload(String cameraId, String contentType, byte[] data, Instant observedAt) {}
}